pkg/queue/sharedmain/main.go

```diff
@@ -304,8 +307,24 @@ func Main(opts ...Option) error {
 	case <-d.Ctx.Done():
 		logger.Info("Received TERM signal, attempting to gracefully shutdown servers.")
 		logger.Infof("Sleeping %v to allow K8s propagation of non-ready state", drainSleepDuration)
+		time.Sleep(drainSleepDuration)
```

Comment:
we don't need this explicit sleep - this is what `drainer.Drain` is doing `drainSleepDuration` is just set to 30s or so